lib/Core/Executor.cpp

```diff
@@ -2566,16 +2576,92 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
     break;
   }
 #endif
+  case Instruction::InsertElement: {
+    InsertElementInst *iei = cast<InsertElementInst>(i);
+    ref<Expr> vec = eval(ki, 0, state).value;
+    ref<Expr> newElt = eval(ki, 1, state).value;
+    ref<Expr> idx = eval(ki, 2, state).value;
+
+    if (!isa<ConstantExpr>(idx)) {
+      terminateStateOnError(
+          state, "InsertElement, support for symbolic index not implemented",
+          Unhandled);
+      return;
+    }
+    ConstantExpr *cIdx = cast<ConstantExpr>(idx);
+    uint64_t iIdx = cIdx->getZExtValue();
 
-  // Other instructions...
-  // Unhandled
-  case Instruction::ExtractElement:
-  case Instruction::InsertElement:
+    const llvm::VectorType *vt = iei->getType();
+    unsigned EltBits = getWidthForLLVMType(vt->getElementType());
+
+    if (iIdx >= vt->getNumElements()) {
+      // Out of bounds write
+      terminateStateOnError(state, "Out of bounds write when inserting element",
+                            BadVectorAccess);
+      return;
+    }
+
+    unsigned ElemCount = vt->getNumElements();
+    ref<Expr> *elems = new ref<Expr>[ElemCount];
+    for (unsigned i = 0; i < ElemCount; ++i) {
+      // evalConstant() will use ConcatExpr to build vectors with the
+      // zero-th element left most (most significant bits), followed
+      // by the next element (second left most) and so on. This means
+      // that we have to adjust the index so we read left to right
+      // rather than right to left.
+      unsigned bitOffset = EltBits*(vt->getNumElements() -i -1);
```

Comment:
Yes we should reuse `ElemCount` (it should be const too). The formatting needs to be fixed too.